resources/manager/data.go

```diff
@@ -88,78 +75,72 @@ func (d *Data) fetchAndSleep(ctx context.Context) {
 	}
 }
 
-// fetchIteration waits for all the registered fetchers and sends all the resources on the output channel.
+// fetchIteration waits for all the registered fetchers and trigger them to fetch relevant resources.
 // The function must not get called in parallel.
 func (d *Data) fetchIteration(ctx context.Context) {
 	d.log.Infof("Manager triggered fetching for %d fetchers", len(d.fetchers.Keys()))
 
 	d.wg = &sync.WaitGroup{}
-	mu := sync.Mutex{}
-	cycleData := make(ResourceMap)
 	start := time.Now()
 
+	cycleId, _ := uuid.NewV4()
+	cycleMetadata := fetching.CycleMetadata{CycleId: cycleId}
+	d.log.Infof("Cycle %s has started", cycleId.String())
+
 	for _, key := range d.fetchers.Keys() {
 		d.wg.Add(1)
 		go func(k string) {
 			defer d.wg.Done()
-			val, err := d.fetchSingle(ctx, k)
+			err := d.fetchSingle(ctx, k, cycleMetadata)
 			if err != nil {
 				d.log.Errorf("Error running fetcher for key %s: %v", k, err)
-			} else if val != nil {
-				d.log.Debugf("Fetcher %s finished and found %d values", k, len(val))
-				mu.Lock()
-				defer mu.Unlock()
-				cycleData[k] = val
 			}
 		}(key)
 	}
 
 	d.wg.Wait()
 	d.log.Infof("Manager finished waiting and sending data after %d milliseconds", time.Since(start).Milliseconds())
-	d.output <- cycleData
+	d.log.Infof("Cycle %s has ended", cycleId.String())
```

Comment:
The Cycle has not really finished, only the task of fetching all resources by `fetchers`.
I wonder If we're fine with sending events after logging this.